### PR TITLE
Delta Diff MVP: Add endpoint

### LINKED ISSUE #4967: Delta Diff MVP - Task: Endpoint
Add an endpoint to the swagger file and an implementation to the controller.

- [x] Lean version
- [ ] Fully functional
- [ ] Tests

### PR DESCRIPTION
**Pull request by Jonathan-Rosenberg**
Add endpoint for the Delta OTF diffs.

## Notice
As detailed in previous PRs, the versions that are **not** merged to `master` are incomplete by design. 
That means that, once again, no tests.

Closes #4967

### CODE REVIEW

**Review by johnnyaug — APPROVED** [review]

File: docs/assets/js/swagger.yml [hunk 1]
```diff
@@ -3868,6 +3952,21 @@ paths:
                 $ref: "#/components/schemas/StorageConfig"
         401:
           $ref: "#/components/responses/Unauthorized"
+  /config/garbage-collection:
```
Comment (talSofer) [inline docs/assets/js/swagger.yml:3955]: how is this related to this pr?

File: pkg/api/controller.go [hunk 2]
```diff
@@ -3854,6 +3854,78 @@ func (c *Controller) PostStatsEvents(w http.ResponseWriter, r *http.Request, bod
...
 }
 
+func (c *Controller) OtfDiff(w http.ResponseWriter, r *http.Request, repository string, leftRef string, rightRef string, params OtfDiffParams) {
+	ctx := r.Context()
+	user, err := auth.GetUser(ctx)
+	if err != nil {
+		writeError(w, r, http.StatusUnauthorized, ErrAuthenticatingRequest)
+		return
+	}
+	c.LogAction(ctx, "table_format_diff", r, repository, rightRef, leftRef)
```
Comment (talSofer) [inline pkg/api/controller.go:3864]: I would add params.Type to the event name. i.e. "table_format_diff_" +params.Type

File: pkg/api/controller.go [hunk 3]
```diff
@@ -3854,6 +3854,78 @@ func (c *Controller) PostStatsEvents(w http.ResponseWriter, r *http.Request, bod
...
+			Key:      credentials[0].AccessKeyID,
+			Secret:   credentials[0].SecretAccessKey,
+			Endpoint: fmt.Sprintf("https://%s", c.Config.ListenAddress),
+		},
+	}
+
+	entries, err := c.otfDiffService.RunDiff(ctx, params.Type, diffParams)
+	if err != nil {
+		c.Logger.Error(err)
+		if errors.Is(err, diff.ErrOTFNotFound) {
```
Comment (talSofer) [inline pkg/api/controller.go:3889]: does the error include the ref that was not found?

Comment (Jonathan-Rosenberg) [reply pkg/api/controller.go:3889]: no

Comment (talSofer) [reply pkg/api/controller.go]: We wanted to add it to understand if a table was removed or added.
